packages/google_maps_flutter/google_maps_flutter_ios/ios/Classes/GoogleMapMarkerController.m

```diff
@@ -201,11 +194,76 @@ - (UIImage *)extractIconFromData:(NSArray *)iconData
                                                      userInfo:nil];
       @throw exception;
     }
+  } else if ([iconData.firstObject isEqualToString:@"asset"]) {
+    if (iconData.count == 4 || iconData.count == 5) {
+      image = [UIImage imageNamed:[registrar lookupKeyForAsset:iconData[1]]];
+      if ([iconData[2] isEqualToString:@"auto"]) {
+        if (iconData.count == 4) {
+          // Update proper scale information for image object.
+          CGFloat imageScale = [iconData[3] doubleValue];
+          image = [self scaleImage:image scale:imageScale];
+        } else if (iconData.count == 5) {
+          CGFloat screenScale = [[UIScreen mainScreen] scale];
+          // Update proper scale information for image object.
+          image = [self scaleImage:image scale:screenScale];
+          // Create resized image.
+          CGSize size = [FLTGoogleMapJSONConversions sizeFromArray:iconData[4] scale:screenScale];
+          image = [self scaleImage:image size:size];
+        }
+      }
+    } else {
+      NSString *error =
+          [NSString stringWithFormat:@"'asset' should have exactly 3 or 4 arguments. Got: %lu",
+                                     (unsigned long)iconData.count];
+      NSException *exception = [NSException exceptionWithName:@"InvalidBitmapDescriptor"
+                                                       reason:error
+                                                     userInfo:nil];
+      @throw exception;
+    }
+  } else if ([iconData[0] isEqualToString:@"bytes"]) {
+    if (iconData.count == 4 || iconData.count == 5) {
+      @try {
```

Comment:
Can you explain what exception are you trying to catch? Maybe the try-catch should be on the method that does the throw?